mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -294,50 +296,55 @@ private FlagsUpdateStageResult handleUpdatesStagedRetry(CassandraId mailboxId, F
         int retryCount = 0;
         while (retryCount < cassandraConfiguration.getFlagsUpdateMessageMaxRetry() && globalResult.containsFailedResults()) {
             retryCount++;
-            FlagsUpdateStageResult stageResult = retryUpdatesStage(mailboxId, flagUpdateCalculator, globalResult.getFailed());
+            FlagsUpdateStageResult stageResult = retryUpdatesStage(mailboxId, flagUpdateCalculator, globalResult.getFailed()).block();
             globalResult = globalResult.keepSucceded().merge(stageResult);
         }
         return globalResult;
     }
 
-    private FlagsUpdateStageResult retryUpdatesStage(CassandraId mailboxId, FlagsUpdateCalculator flagsUpdateCalculator, List<MessageUid> failed) {
-        Stream<ComposedMessageIdWithMetaData> idsFailed = FluentFutureStream.of(
-                failed.stream().map(uid -> messageIdDAO.retrieve(mailboxId, uid)),
-                FluentFutureStream::unboxOptional)
-            .join();
```

Comment:
I dont see any `.block()` corresponds to `.join()` of your code changes, could you check it?

Reply:
Because I have make this method `Mono` to be able to compose it better.

Reply:
What I meant was, `.join()` in original code shows me that statement is intended to be executed in blocking fashion. you did specify any blocking method in your refactored code. I fear that would change the behavior of the logic.

Reply:
In fact we went from:

    private FlagsUpdateStageResult retryUpdatesStage(...)

to:

    private Mono<FlagsUpdateStageResult> retryUpdatesStage(...)

And on the calling side:

From:

            FlagsUpdateStageResult stageResult = retryUpdatesStage(mailboxId, flagUpdateCalculator, globalResult.getFailed());

To:

            FlagsUpdateStageResult stageResult = retryUpdatesStage(mailboxId, flagUpdateCalculator, globalResult.getFailed()).block();

It is the same intent, the `block()` is just delayed.